src/raftstore/coprocessor/region_info_accessor.rs

```diff
@@ -328,7 +257,93 @@ impl RegionCollector {
         }
     }
 
+    /// Determines whether `region_to_check`'s epoch is stale compared to `current`'s epoch
+    #[inline]
+    fn is_region_epoch_stale(&self, region_to_check: &Region, current: &Region) -> bool {
+        let epoch = region_to_check.get_region_epoch();
+        let current_epoch = current.get_region_epoch();
+
+        epoch.get_version() < current_epoch.get_version()
+            || epoch.get_conf_ver() < current_epoch.get_conf_ver()
+    }
+
+    /// For all regions whose range overlaps with given `regions` or region_id is the same as
```

Comment:
with the given region

Reply:
Check wether the given region's epoch is not stale with all the regions whose range overlap with the given region, or the region_id is the same as the give region.
BTW `checks whether the given `region`'s epoch is not older than theirs` seems not consistency with the function name?